src/qibolab/__init__.py

```diff
@@ -54,3 +55,25 @@ def create_platform(name, runcard=None):
     if runcard is None:
         return module.create()
     return module.create(runcard)
+
+
+def execute_qasm(circuit: str, platform, runcard=None, initial_state=None, nshots=1000):
+    """Executes a QASM circuit.
+
+    Args:
+        circuit (str): the QASM circuit.
+        platform (str): the platform where to execute the circuit.
+        runcard (): the runcard used for the platform.
+        initial_state (:class:`qibo.models.circuit.Circuit`): Circuit to prepare the initial state.
+                If ``None`` the default ``|00...0>`` state is used.
+        nshots (int): Number of shots to sample from the experiment.
+
+    Returns:
+        ``MeasurementOutcomes`` object containing the results acquired from the execution.
+    """
+    from qibolab.backends import QibolabBackend
+
+    circuit = Circuit.from_qasm(circuit)
+    return QibolabBackend(platform, runcard).execute_circuit(
```

Comment:
I would prefer to return the results object after circuit execution instead of the circuit. Otherwise the name of this function is misleading, and most likely the user will prefer to get directly results instead of an intermediate representation.

Reply:
This should be returning directly the results (`MeasurementOutcomes`) since it is actually executing the circuit (it is using `execute_circuit`).